pyroSAR/drivers.py

```diff
@@ -1490,22 +1500,664 @@ def scanMetadata(self):
+#     >>> archive = ArchiveSpatialite('/path/to/dbfile.db')
+#     >>> site = Vector('/path/to/site.shp')
+#     >>> outdir = '/path/to/processed/results'
+#     >>> maxdate = '20171231T235959'
+#     >>> selection_proc = archive.select(vectorobject=site, processdir=outdir,
+#     >>>                                 maxdate=maxdate, sensor=('S1A', 'S1B'),
+#     >>>                                 product='GRD', acquisition_mode='IW', vv=1)
+#     >>> archive.close()
+#
+#     Alternatively, the `with` statement can be used.
+#     In this case to just check whether one particular scene is already registered in the database:
+#
+#     >>> from pyroSAR import identify, ArchiveSpatialite
+#     >>> scene = identify('S1A_IW_SLC__1SDV_20150330T170734_20150330T170801_005264_006A6C_DA69.zip')
+#     >>> with ArchiveSpatialite('/path/to/dbfile.db') as archive:
+#     >>>     print(archive.is_registered(scene.scene))
+#     """
+#
+#     def __init__(self, dbfile, custom_fields=None):
+#         self.dbfile = dbfile
+#         self.conn = sqlite_setup(dbfile, ['spatialite'])
+#
+#         self.lookup = {'sensor': 'TEXT',
+#                        'orbit': 'TEXT',
+#                        'orbitNumber_abs': 'INTEGER',
+#                        'orbitNumber_rel': 'INTEGER',
+#                        'cycleNumber': 'INTEGER',
+#                        'frameNumber': 'INTEGER',
+#                        'acquisition_mode': 'TEXT',
+#                        'start': 'TEXT',
+#                        'stop': 'TEXT',
+#                        'product': 'TEXT',
+#                        'samples': 'INTEGER',
+#                        'lines': 'INTEGER',
+#                        'outname_base': 'TEXT PRIMARY KEY',
+#                        'scene': 'TEXT',
+#                        'hh': 'INTEGER',
+#                        'vv': 'INTEGER',
+#                        'hv': 'INTEGER',
+#                        'vh': 'INTEGER'}
+#
+#         if custom_fields is not None:
+#             self.lookup.update(custom_fields)
+#
+#         create_string = '''CREATE TABLE if not exists data ({})'''.format(
+#             ', '.join([' '.join(x) for x in self.lookup.items()]))
+#         cursor = self.conn.cursor()
+#         cursor.execute(create_string)
+#         if 'bbox' not in self.get_colnames():
+#             cursor.execute('SELECT AddGeometryColumn("data","bbox" , 4326, "POLYGON", "XY", 0)')
+#
+#         create_string = 'CREATE TABLE if not exists duplicates (outname_base TEXT, scene TEXT)'
+#         cursor.execute(create_string)
+#         self.conn.commit()
+#         sys.stdout.write('\rchecking for missing scenes..')
+#         self.cleanup()
+#         sys.stdout.write('\rchecking for missing scenes..done\n')
+#         sys.stdout.flush()
+#
+#     def __prepare_insertion(self, scene):
+#         """
+#         read scene metadata and parse a string for inserting it into the database
+#
+#         :param scene: a SAR scene
```

Comment:
please change to numpy-style docstrings to be consistent with the rest of the code